packages/shared/src/modals/FileUploadInput.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { getValidatedProp } from '@odf/shared/utils';
+import { TFunction } from 'i18next';
+import { FileUpload, FormGroup } from '@patternfly/react-core';
+import { KMSMaxFileUploadSize } from '../../../odf/constants';
+import { AzureConfig, ThalesConfig } from '../../../odf/types';
+
+export const uploadInputItems: UploadInputItemType = (t) => [
+  {
+    id: 'clientCert',
+    inputName: t('Client certificate'),
+  },
+  {
+    id: 'caCert',
+    inputName: t('CA certificate'),
+  },
+  {
+    id: 'clientKey',
+    inputName: t('Client private key'),
+  },
+];
+
+export const FileUploadInput: React.FC<
+  FileUploadInputProps<AzureConfig | ThalesConfig>
+> = ({
+  key,
+  id,
+  inputName,
+  helperText,
+  labelIcon,
+  className,
+  kmsState,
```

Comment:
why have you removed the props which were originally passed to this component ??
```suggestion
  kmsState,
  kmsStateClone,
```

Reply:
check the original implementation again and use correct objects (kmsState & kmsStateClone) at required places...

Reply:
Sure